File: Data Structures & Algorithms/trapping-rain-water/submission-0.js
class Solution {
    /**
     * @param {number[]} height
     * @return {number}
     */
    trap(height) {
        let ans=0;
        let l=0;
        let r=height.length-1
        let maxL=height[0];
        let maxR=height[height.length-1]
        //for(let i=0;i<height.length;i++){
        while(l<r){
            if(maxL<maxR){
                l+=1
                maxL=Math.max(maxL,height[l]);
                ans+=maxL-height[l];
                // if(sum>0){
                //     ans+=sum
                // }
                // if(height[l]>maxL){
                //     maxL=height[l]
                // }
                
            }
            else{
                r-=1
                maxR=Math.max(maxR, height[r]);
                ans+=maxR-height[r]
                // if(sum>0){
                //     ans+=sum
                // }
                // if(height[r]>maxR){
                //     maxR=height[r]
                // }
                
            }
        }
        return ans
    }
}

File: Data Structures & Algorithms/house-robber/submission-0.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        nums.append(0)
        i=len(nums)-2
        while i>-1:
            if i+2 < len(nums) and i+3 <len(nums):
                nums[i]+=max(nums[i+2], nums[i+3])
            i-=1
        return max(nums[0],nums[1])